server/src/main/java/io/whitefox/core/JsonPredicates.java

```diff
@@ -0,0 +1,385 @@
+package io.whitefox.core;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+class EvalContext {
+
+    public EvalContext(Map<String, String> partitionValues, Map<String, Pair<String, String>> statsValues) {
+        this.partitionValues = partitionValues;
+        this.statsValues = statsValues;
+    }
+
+    Map<String, String> partitionValues;
+    Map<String, Pair<String, String>> statsValues;
+}
+
+class OpDataTypes {
+    static final String BoolType = "bool";
+    static final String IntType = "int";
+    static final String LongType = "long";
+    static final String StringType = "string";
+    static final String DateType = "date";
+    static final String FloatType = "float";
+    static final String DoubleType = "double";
+    static final String TimestampType = "timestamp";
```

Comment:
can we use the types defined in `io.whitefox.core.types`?

Reply:
ill check it out, probably